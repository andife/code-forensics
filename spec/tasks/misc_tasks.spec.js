/* eslint-disable jest/expect-expect */
var Bluebird = require('bluebird');

var miscTasks = require('tasks/misc_tasks');

var taskHelpers = require('../jest_tasks_helpers');

describe('Misc Tasks', function() {
  describe('generate-layer-grouping-files', function() {
    var runtime;
    var contextConfig = {
      layerGroups: {
        'test_group': [
          { name: 'Test Layer1', paths: ['test/path1', /test\/path2\/((?!.*--abc\.)).*\/files/] },
          { name: 'Test Layer2', paths: ['test_path3', /^test\/path4\/.*\.cf$/] }
        ]
      }
    };

    afterEach(function() {
      return runtime.clear();
    });

    describe('with the layer group parameter', function() {
      var assertLayerFiles = function(taskOutput) {
        return Bluebird.all([
          taskOutput.assertTempFile('layer-groups.txt'),
          taskOutput.assertTempFile('layer-group-test-layer-1.txt'),
          taskOutput.assertTempFile('layer-group-test-layer-2.txt')
        ]);
      };

      beforeEach(function() {
        runtime = taskHelpers.createRuntime('misc_tasks', miscTasks, contextConfig, { layerGroup: 'test_group' });
      });

      describe('as a Task', function() {
        it('generates layer grouping files', function() {
          return runtime.executePromiseTask('generate-layer-grouping-files')
            .then(assertLayerFiles);
        });
      });

      describe('as a Function', function() {
        it('generates layer grouping files', function() {
          return runtime.executePromiseFunction('generateLayerGroupingFiles')
            .then(assertLayerFiles);
        });
      });
    });

    describe('with no layer group parameter', function() {
      var assertNoLayerFiles = function(taskOutput) {
        return taskOutput.assertMissingTempReports([
          'layer-groups.txt',
          'layer-group-test-layer-1.txt',
          'layer-group-test-layer-2.txt'
        ]);
      };

      beforeEach(function() {
        runtime = taskHelpers.createRuntime('misc_tasks', miscTasks, contextConfig);
      });

      describe('as a Task', function() {
        it('does not generate layer grouping files', function() {
          return runtime.executePromiseTask('generate-layer-grouping-files')
            .then(assertNoLayerFiles);
        });
      });

      describe('as a Function', function() {
        it('does not generate a layer grouping file', function() {
          return runtime.executePromiseFunction('generateLayerGroupingFiles')
            .then(assertNoLayerFiles);
        });
      });
    });
  });
});
