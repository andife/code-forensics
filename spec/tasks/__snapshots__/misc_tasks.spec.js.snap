// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Misc Tasks generate-layer-grouping-files with the layer group parameter as a Function generates layer grouping files: layer-group-test-layer-1.txt 1`] = `
"test/path1 => Test Layer1
^test\\\\/path2\\\\/((?!.*--abc\\\\.)).*\\\\/files$ => Test Layer1"
`;

exports[`Misc Tasks generate-layer-grouping-files with the layer group parameter as a Function generates layer grouping files: layer-group-test-layer-2.txt 1`] = `
"test_path3 => Test Layer2
^test\\\\/path4\\\\/.*\\\\.cf$ => Test Layer2"
`;

exports[`Misc Tasks generate-layer-grouping-files with the layer group parameter as a Function generates layer grouping files: layer-groups.txt 1`] = `
"test/path1 => Test Layer1
^test\\\\/path2\\\\/((?!.*--abc\\\\.)).*\\\\/files$ => Test Layer1
test_path3 => Test Layer2
^test\\\\/path4\\\\/.*\\\\.cf$ => Test Layer2"
`;

exports[`Misc Tasks generate-layer-grouping-files with the layer group parameter as a Task generates layer grouping files: layer-group-test-layer-1.txt 1`] = `
"test/path1 => Test Layer1
^test\\\\/path2\\\\/((?!.*--abc\\\\.)).*\\\\/files$ => Test Layer1"
`;

exports[`Misc Tasks generate-layer-grouping-files with the layer group parameter as a Task generates layer grouping files: layer-group-test-layer-2.txt 1`] = `
"test_path3 => Test Layer2
^test\\\\/path4\\\\/.*\\\\.cf$ => Test Layer2"
`;

exports[`Misc Tasks generate-layer-grouping-files with the layer group parameter as a Task generates layer grouping files: layer-groups.txt 1`] = `
"test/path1 => Test Layer1
^test\\\\/path2\\\\/((?!.*--abc\\\\.)).*\\\\/files$ => Test Layer1
test_path3 => Test Layer2
^test\\\\/path4\\\\/.*\\\\.cf$ => Test Layer2"
`;
